File: src/python/_0038_Count_and_Say.py
class Solution:
    def countAndSay(self, n: int) -> str:
        if n == 1:
            return "1"

        ans = ""
        lastAns = self.countAndSay(n - 1)
        seenCh = lastAns[0]
        freq = 0

        for ch in lastAns:
            if seenCh == ch:
                freq += 1
            else:
                ans += f"{freq}{seenCh}"
                freq = 1
                seenCh = ch

        ans += f"{freq}{seenCh}"

        return ans

File: src/_0038_Count_and_Say.java
class Solution {
    public String countAndSay(int n) {
        if (n == 1)
            return "1";

        StringBuilder sb = new StringBuilder();
        String lastAns = this.countAndSay(n - 1);
        char seenCh = lastAns.charAt(0);
        int freq = 1;

        for (int i = 1; i < lastAns.length(); i++) {
            char ch = lastAns.charAt(i);
            if (seenCh == ch) {
                freq++;
            } else {
                sb.append(freq).append(seenCh);
                freq = 1;
                seenCh = ch;
            }
        }

        sb.append(freq).append(seenCh);

        return sb.toString();
    }
}
